### ISSUE TITLE
Is there no option to customize the code block tags?

### ISSUE THREAD

**Issue by philipwalton** [jonschlinkert/remarkable#131]
When I override the `highlight` option, remarkable still wraps what I return in the following tags:

```html
<pre><code class="language-css">...</code></pre>
```

As far as I can tell, the only customization option I have is `langPrefix`. I was expecting to be able to override everything when passing a `highlight` function.

**Comment by KyleAMathews** [comment]
Agreed. Can't get highlight.js css working right now as there's no `.hljs` class set.

**Comment by k88hudson** [comment]
I did this https://github.com/k88hudson/remarkable/commit/65a7990f2027191caacd3b111def650e87995c5d which works for me for now, but I'm assuming there's a better way to extend the render function here (I didn't spend a lot time reading through the source). Would love to hear about it when you get around to this issue @jonschlinkert ^_^

**Comment by k88hudson** [comment]
Ah ok... I read the code a bit more, and it seems you're already checking for a `fence_custom` rule, which works nicely :+1:

It *is* kind of a lot of code to copy just to change the HTML structure though, having the option to override just that part would be kind of nice.

**Comment by rarkins** [comment]
I've now experienced the exact same problem. I would go as far as to say that the current `highlight` implementation in remarkable is not fully compatible with `highlight.js` when used on the server side because `highlight` CSS needs the `hljs` class on the `<pre>` element, as @KyleAMathews noted.

It could be nice to have an explicit (named) highlight type the same way you have `full` and `commonmark` defined at the top level. In that case perhaps the class `hljs` can be added to every instance of `<pre><code` that's found.

Also a problem with the recommended code for `highlight.js` is that it doesn't provide any way to escape highlighting, e.g. `nohighlight`, but that's easily overcome as that part is your own code.

**Comment by rarkins** [comment]
Actually I think the `hljs` class may need to be on the `<code>` element, not `<pre>`. Example from highlightjs.org demo page, after JS:
![image](https://cloud.githubusercontent.com/assets/6311784/15406101/46cb9908-1e05-11e6-905a-86a8cb7d50f8.png)

**Comment by mootari** [comment]
A simple plugin to provide a fallback language:
```javascript
/**
 * Markdown plugin, sets fallback language for fenced code blocks.
 */
function codeDefaultLang(md) {
  const rule = md.renderer.rules.fence;
  md.renderer.rules.fence = function(tokens, idx, options, env, instance) {
    if(!tokens[idx].params && md.options.langDefault) {
      tokens[idx].params = md.options.langDefault;
    }
    return rule.call(this, tokens, idx, options, env, instance);
  };
}
```
Usage:
```javascript
const options = {
  // highlight and other options ...
  langPrefix: 'hljs language-',
  langDefault: 'unknown'
};
new Remarkable().use(codeDefaultLang);
```

**Comment by domleonard** [comment]
I've encountered this issue in 2021 - stylesheets supplied with highlight.js  only half work because their background rules are for a `.hljs` class block container - and the `<pre>` tag  is hard coded in the return statement of [`rules.fence`](https://github.com/jonschlinkert/remarkable/blob/master/lib/rules.js) with no class name.

The work around below was written to insert the class name on the `<pre>` tag in a wrapper function around a call to the original  `rules.fence` code  (which would cause an infinite loop if done in a  `fence_custom` rule):


```js
// add .hljs class to <pre> tag in a plugin   *** when highlight.js is being used **
const fenceCode = md.renderer.rules.fence;
md.renderer.rules.fence = function (...args) {
    return fenceCode.call( md.renderer.rules, ...args).replace( '<pre>', '<pre class="hljs">');
};
``` 
While this is working well (one can suddenly see what themes look like) it is a bit of a hack and I do think providing a way of modifying the return tags of `rules.fence` in options or some kind of customization rule would be a cleaner solution.
